compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/AggregatedOpInterfaceImpl.cpp

```diff
@@ -503,9 +526,9 @@ FailureOr<SmallVector<Value>> Im2colOp::decomposeOperation(OpBuilder &b) {
       break;
     }
   }
-  bool tileK =
-      !willBeContiguousSlice(innerSliceSize, kTileSize, kOffset.front());
-  if (!tileK) {
+  bool vectorizeInnerKLoop =
+      willBeContiguousSlice(innerSliceSize, kTileSize, kOffset);
```

Comment:
don't we need to take basis into account?

Reply:
Having an inner basis of greater than 1 would mean that the insert_slices into the output tensor are strided, and I haven't really thought through what the semantics of that should be, since it shouldn't ever show up in the IR. I'm also pretty sure the decomposition would not generate correct IR for this case anyway, so I am going to restrict the inner basis to always be 1 in the decomposition and verifier. Nice catch!